src/Python/Meshes/Decimation.py

```diff
@@ -0,0 +1,180 @@
+    args = parser.parse_args()
+    return args.filename, args.reduction
+
+
+def main():
+    filePath, reduction = get_program_parameters()
+    
+    # Define colors
+    colors = vtk.vtkNamedColors()
+    backFaceColor = colors.GetColor3d("gold")
+    inputActorColor = colors.GetColor3d("flesh")
+    decimatedActorColor = colors.GetColor3d("flesh")
+
+    if filePath and os.path.isfile(filePath):
+        inputPolyData = ReadPolyData(filePath)
+        if not inputPolyData:
+            inputPolyData = GetSpherePD()
+    else:
+        inputPolyData = GetSpherePD()
+
+    if not reduction:
+        reduction = .9 # 90% reduction
+
+    print("Before decimation")
+    print(f"There are {inputPolyData.GetNumberOfPoints()} points.")
+    print(f"There are {inputPolyData.GetNumberOfPolys()} polygons.")
+
+    decimate = vtk.vtkDecimatePro()
+    decimate.SetInputData(inputPolyData)
+    decimate.SetTargetReduction(reduction)
+    decimate.PreserveTopologyOn()
+    decimate.Update()
+
+    decimated = vtk.vtkPolyData()
+    decimated.ShallowCopy(decimate.GetOutput())
+
+    print("After decimation")
+    print(f"There are {decimated.GetNumberOfPoints()} points.")
+    print(f"There are {decimated.GetNumberOfPolys()} polygons.")
+    print(f"Reduction: {(inputPolyData.GetNumberOfPolys() - decimated.GetNumberOfPolys()) / inputPolyData.GetNumberOfPolys()}")
+
+    inputMapper = vtk.vtkPolyDataMapper()
+    inputMapper.SetInputData(inputPolyData)
+
+    backFace = vtk.vtkProperty()
+    backFace.SetColor(backFaceColor)
+
+    inputActor = vtk.vtkActor()
+    inputActor.SetMapper(inputMapper)
+    inputActor.GetProperty().SetInterpolationToFlat()
+    inputActor.GetProperty().SetColor(inputActorColor)
+    inputActor.SetBackfaceProperty(backFace)
+
+    decimatedMapper = vtk.vtkPolyDataMapper()
+    decimatedMapper.SetInputData(decimated)
+
+    decimatedActor = vtk.vtkActor()
+    decimatedActor.SetMapper(decimatedMapper)
+    decimatedActor.GetProperty().SetColor(decimatedActorColor)
+    decimatedActor.GetProperty().SetInterpolationToFlat()
+    decimatedActor.SetBackfaceProperty(backFace)
+
+    # There will be one render window
+    renderWindow = vtk.vtkRenderWindow()
+    renderWindow.SetSize(600, 300)
+
+    # And one interactor
+    interactor = vtk.vtkRenderWindowInteractor()
+    interactor.SetRenderWindow(renderWindow)
+
+    # Define viewport ranges
+    # (xmin, ymin, xmax, ymax)
+    leftViewport = [0.0, 0.0, 0.5, 1.0]
+    rightViewport = [0.5, 0.0, 1.0, 1.0]
+
+    # Setup both renderers
+    leftRenderer = vtk.vtkRenderer()
+    renderWindow.AddRenderer(leftRenderer)
+    leftRenderer.SetViewport(leftViewport)
+    leftRenderer.SetBackground(.6, .5, .4)
+
+    rightRenderer = vtk.vtkRenderer()
+    renderWindow.AddRenderer(rightRenderer)
+    rightRenderer.SetViewport(rightViewport)
+    rightRenderer.SetBackground(.4, .5, .6)
+
```

Comment:
Replace this line with: `rightRenderer.SetBackground(colors.GetColor3d('rightBkg'))`